x-pack/plugins/security_solution/server/lib/telemetry/endpoint_metadata_telemetry.ts

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { ISavedObjectsRepository } from 'src/core/server';
+import { AgentMetadata } from './../../../../ingest_manager/common/types/models/agent';
+import { AGENT_SAVED_OBJECT_TYPE } from './../../../../ingest_manager/common/constants/agent';
+import { AgentSOAttributes } from '../../../../ingest_manager/common';
+
+export interface EndpointMetadataOSTelemetry {
+  name: string;
+  version: string;
+  count: number;
+}
+
+export interface EndpointPoliciesTelemetry {
+  malware: {
+    success: number;
+    warning: number;
+    failure: number;
+  };
+}
+
+export interface IngestManagerLocalMetadata extends AgentMetadata {
+  host: {
+    id: string;
+  };
+  os: {
+    name: string;
+    version: string;
+    full: string;
+  };
+}
+
+export interface EndpointMetadataTelemetry {
+  endpoints: {
+    installed: number;
+    last_24_hours: number;
+  };
+  os: Record<string, EndpointMetadataOSTelemetry>;
+  policies: EndpointPoliciesTelemetry;
+}
+
+export const getDefaultEndpointMetadataTelemetry = (): EndpointMetadataTelemetry => ({
+  endpoints: { installed: 0, last_24_hours: 0 },
+  os: {} as Record<string, EndpointMetadataOSTelemetry>,
+  policies: {
+    malware: {
+      success: 0,
+      warning: 0,
+      failure: 0,
+    },
+  },
+});
+
+/**
+ * ---Functionality Tracking---
+ * Number of active endpoints (in progress)
+ *  - Successfully installed (done)
+ *  - Has sent response within last 24 hours (best way to go about this? May not have info?)
+ * See the list by customer (provided by telemetry under license.issued_to) Can most likely pull those details in the dashboards
+ * OS (done)
+ *  - Type (done)
+ *  - Version (done)
+ * Geolocation of endpoints (slated for 7.10): Query the agent based on the agent info in the metadata to retrieve this
+ * ---------------------------------
+ * RE Performance / usage collection timeout from Tina Helligers
+ * We have a timeout on waiting for collectors to report they are ready before issuing a bulkFetch method that essentially calls the fetch method for all the collectors registered.
+ * This timeout defaults to 60s. If you suspect that your fetch method is going to take too long (meaning data for that collector won't be reported),
+ * you could follow a similar approach as what we do for oss_telemetry where we use the task manager to run on a daily basis.
+ */
+
+const ENDPOINT_PACKAGE_CONSTANT = 'system'; // TODO: replace with 'endpoint' when I have endpoint package working remotely
+
+export const getEndpointMetadataTelemetryFromFleet = async (
+  savedObjectsClient: ISavedObjectsRepository
+) => {
+  const { saved_objects: endpointAgents } = await savedObjectsClient.find<AgentSOAttributes>({
```

Comment:
@michaelolo24 are these SOs global or space-aware? If you need to search across spaces you'll need to query the kibana index directly. You can see an example of that in [rules/jobs telemetry](https://github.com/elastic/kibana/pull/71102/files#diff-4c22c40434217b971d4602ff957541ecR99-R110).

Reply:
They are globals? I was told that I could access the SO directly if they were listed as `namepaceType: 'agnostic'`. I'll take a look